packages/transformers/babel/src/config.js

```diff
@@ -1,141 +1,191 @@
+      createConfigItem(plugin, {type: 'plugin', dirname: BABEL_TRANSFORMER_DIR})
+    );
+    config.shouldRehydrate();
+    prepForReyhdration(babelOptions);
   }
 
-  return result;
+  config.setResult({
+    internal: true,
+    config: babelOptions
+  });
+  await definePluginDependencies(config);
 }
 
-function mergeConfigs(result, config) {
+function mergeOptions(result, config?: null | BabelConfig) {
   if (
     !config ||
-    ((!config.config.presets || config.config.presets.length === 0) &&
-      (!config.config.plugins || config.config.plugins.length === 0))
+    ((!config.presets || config.presets.length === 0) &&
+      (!config.plugins || config.plugins.length === 0))
   ) {
-    return;
+    return result;
   }
 
-  let merged = result[config.babelVersion];
+  let merged = result;
   if (merged) {
-    merged.config.presets = (merged.config.presets || []).concat(
-      config.config.presets || []
-    );
-    merged.config.plugins = (merged.config.plugins || []).concat(
-      config.config.plugins || []
-    );
+    merged.presets = (merged.presets || []).concat(config.presets || []);
+    merged.plugins = (merged.plugins || []).concat(config.plugins || []);
   } else {
-    result[config.babelVersion] = config;
+    result = config;
   }
+
+  return result;
 }
 
-function hasPlugin(arr, plugins) {
-  return (
-    Array.isArray(arr) && arr.some(p => plugins.includes(getPluginName(p)))
+function getStats(options) {
+  let canBeRehydrated = true;
+  let dependsOnRelative = false;
+  let dependsOnLocal = false;
+
+  let configItems = [...options.presets, ...options.plugins];
+
+  for (let configItem of configItems) {
+    if (!configItem.file) {
+      canBeRehydrated = false;
+    } else if (configItem.file.request.startsWith('.')) {
+      dependsOnRelative = true;
+    } else if (isLocal(/*configItem.file.resolved*/)) {
+      dependsOnLocal = true;
+    }
+  }
+
+  return {canBeRehydrated, dependsOnRelative, dependsOnLocal};
+}
+
+async function isExtended(/* babelrcPath */) {
+  // TODO: read and parse babelrc and check to see if extends property exists
+  // need access to fs in case of memory filesystem
+  return false;
+}
+
+async function isLocal(/* configItemPath */) {
+  // TODO: check if realpath is different, need access to fs in case of memory filesystem
+  return false;
+}
+
+function prepForReyhdration(options) {
+  options.presets = (options.presets || []).map(configItem => ({
+    file: configItem.file,
+    options: configItem.options
+  }));
+  options.plugins = (options.plugins || []).map(configItem => ({
+    file: configItem.file,
+    options: configItem.options
+  }));
+}
+
+async function definePluginDependencies(config) {
+  let babelConfig = config.result.config;
+  if (babelConfig == null) {
+    return;
+  }
+
+  let configItems = [...babelConfig.presets, ...babelConfig.plugins];
+  await Promise.all(
+    configItems.map(async configItem => {
+      let pkg = nullthrows(
```

Comment:
Does the request graph already handle resolving these package versions?

Reply:
The request graph now skips resolving dev dep versions if it sees that they were already resolved by the plugin.

Reply:
Sure, but if it's already being handled by the graph do we need to do it here?

Reply:
Ah yeah. I was thinking that Babel already did the work of resolving the package (and then the entry file for that package) so it would be less work to resolve the `package.json` from within the package rather than resolving all over again.